TestHelper/MemoryLogger.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.TestHelper
+{
+    using NLog;
+    using NLog.Config;
+    using NLog.Targets;
+
+    using System.Diagnostics;
+
+    public static class MemoryLogger
+    {
+        public static (Extensions.Logging.ILogger logger, MemoryTarget logTarget) Create()
+        {
+            // Use reflection to get the name of the class which declares our caller.
+            // Expected to be the name of a Test class.
+            StackFrame frame = new StackFrame(1);

```

Comment:
I was looking into the `StackFrame` constructor and it was `Initializes a new instance of the [StackFrame](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.stackframe?view=net-6.0) class that corresponds to a frame above the current stack frame.` We choose this value to be 1. Basically this is the reflection we use to get the name of our class?

Reply:
Yep! The (1) goes up 1 stack frame, which is generally going to be a Unit test class.